### ISSUE TITLE
Incorrect "IDE0001 Name can be simplified" diagnostic on switch statement returning expression tree, code fix produces invalid code

### ISSUE THREAD

**Issue by thomaslevesque** [dotnet/roslyn#41006]
**Version Used**: 
8.0

**Steps to Reproduce**:

1. Write the following code in Visual Studio 2019 (16.4.2)
    ```csharp
    Expression<Func<Foo, bool>> filter = (_fooId, _fooOtherId) switch
    {
        (Guid fooId, _) => foo => foo.Id == _fooId.Value,
        (null, string s) when s.Length > 0 => foo => foo.OtherId == _fooOtherId,
        _ => null
    };
    ```

    (actual code, with only types and variable names changed)

**Expected Behavior**:

No diagnostic should be reported for `Expression<Func<Foo, bool>>`

**Actual Behavior**:

Two diagnostics are reported for `Expression<Func<Foo, bool>>`:
- IDE0001 (Name can be simplified)
- IDE0007 (Use var instead of explicit type)

Applying the code fix for either of these diagnostics changes `Expression<Func<Foo, bool>>` to `var`, which is illegal, since lambda expressions don't have an intrinsic type.

**Comment by CyrusNajmabadi** [comment]
Fully compiling example:

```c#
#nullable disable

using System;
using System.Linq.Expressions;

class Foo { public Guid Id; public object OtherId; }

class Example
{
    Guid? _fooId;
    object _fooOtherId;

    public void Method()
    {
        Expression<Func<Foo, bool>> filter = (_fooId, _fooOtherId) switch
        {
            (Guid fooId, _) => foo => foo.Id == _fooId.Value,
            (null, string s) when s.Length > 0 => foo => foo.OtherId == _fooOtherId,
            _ => null
        };
    }
}
```

**Comment by CyrusNajmabadi** [comment]
Closing out as i get neither IDE0001 nor IDE0007 with this code.
